dsgrid/registry/project_registry_manager.py

```diff
@@ -267,6 +280,114 @@ def _register_project_and_dimensions(
+        )
+        # Order of the next two is required for Pydantic validation.
+        dimension_references += self._dimension_mgr.make_dimension_references(dimension_ids)
+        dimensions.clear()
+
+    def _register_all_in_one_dimensions(
+        self,
+        src_dir,
+        model,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        new_dimensions = []
+        new_mappings = {}
+        dim_type_to_ref = {x.dimension_type: x for x in model.dimensions.base_dimension_references}
+        for dimension_type in model.dimensions.all_in_one_supplemental_dimensions:
+            if dimension_type == DimensionType.TIME:
+                raise DSGInvalidParameter("Cannot have time in all_in_one_supplemental_dimensions")
+            if dimension_type in new_mappings:
+                raise DSGInvalidParameter(
+                    f"{dimension_type} is listed twice in all_in_one_supplemental_dimensions"
+                )
+
+            dim_ref = dim_type_to_ref[dimension_type]
+            dim_config = self._dimension_mgr.get_by_id(dim_ref.dimension_id)
+            supp_dim_dir = src_dir / "dimensions" / "supplemental"
+            if not supp_dim_dir.exists():
+                raise DSGInvalidParameter(
+                    f"supplemental dimensions directory {supp_dim_dir} does not exist"
+                )
+            supp_map_dir = src_dir / "dimension_mappings" / "base_to_supplemental"
+            if not supp_map_dir.exists():
+                raise DSGInvalidParameter(
+                    f"supplemental dimension mappings directory {supp_map_dir} does not exist"
+                )
+            dt_str = dimension_type.value
+            dt_plural = f"all_{dt_str}s"
+            dt_plural_dash = f"all-{dt_str}s"
+            dt_plural_formal = f"All {dt_str[0].upper()}{dt_str[1:]}"
+            dim_record_file = supp_dim_dir / f"{dt_plural}.csv"
+            dim_text = f"id,name\nall_{dt_str}s,{dt_plural_formal}\n"
+            dim_record_file.write_text(dim_text)
+            map_record_file = supp_map_dir / f"lookup_{dt_str}_to_{dt_plural}.csv"
+            with open(map_record_file, "w") as f_out:
+                f_out.write("from_id,to_id\n")
+                for record in dim_config.get_unique_ids():
+                    f_out.write(record)
+                    f_out.write(",")
+                    f_out.write(dt_plural)
+                    f_out.write("\n")
+
+            new_dim = run_in_other_dir(
+                src_dir,
+                DimensionModel,
+                filename=str(Path("dimensions") / "supplemental" / dim_record_file.name),
+                name=dt_plural_dash,
+                display_name="None",
+                dimension_type=dimension_type,
+                module=dim_config.model.module,
+                class_name=dim_config.model.class_name,
+                description=dt_plural_formal,
+            )
+            new_dimensions.append(new_dim)
```

Comment:
I don't get exactly what this run_in_other_dir function is doing, but I trust you that it's making the new dimension ...

Reply:
The Pydantic model that performs validation checks file paths. The paths in the model are relative to `src_dir`. The way we typically make that work is to run the validation from that directory.

Reply:
Makes sense. Thanks for explaining!